pkg/pillar/cmd/downloader/space.go

```diff
@@ -26,10 +26,15 @@ func initSpace(ctx *downloaderContext, kb uint64) {
 // Returns true if there was space
 func tryReserveSpace(ctx *downloaderContext, status *types.DownloaderStatus,
 	kb uint64) bool {
+	if status.ReservedSpace != 0 {
+		log.Errorf("%s, space is already reserved\n", status.Name)
+		return true
+	}
 
 	ctx.globalStatusLock.Lock()
 	if kb >= ctx.globalStatus.RemainingSpace {
 		ctx.globalStatusLock.Unlock()
+		log.Errorf("%s, space can not be reserved\n", status.Name)
```

Comment:
Wouldn't it make sense to print kb and RemainingSpace in the error message?

Reply:
Sure, done.